pkg/kube/client.go

```diff
@@ -520,7 +520,8 @@ func podsReady(pods []api.Pod) bool {
 
 func servicesReady(svc []api.Service) bool {
 	for _, s := range svc {
-		if !api.IsServiceIPSet(&s) {
+		// Make sure the service is not explicitly set to "None" before checking the IP
+		if s.Spec.ClusterIP != api.ClusterIPNone && !api.IsServiceIPSet(&s) {
```

Comment:
I think that the helper function checks this already
https://github.com/kubernetes/kubernetes/blob/v1.5.4/pkg/api/helpers.go#L236-L240